Tema-Java8/src/main/java/java8/Java8Main.java

```diff
@@ -0,0 +1,278 @@
+package java8;
+
+import javax.swing.text.html.Option;
+import java.awt.*;
+import java.util.*;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Java8Main {
+    //I made it 'global'. that's the only way I know to use it in 2 or more methods :(
+    public static Set<String> randomString = new HashSet<>();
+
+    public static void main(String[] args) {
+
+        // 1) Create and Add 5 different instances of an object in a list.
+        List<Flower> flowerList = Arrays.asList(
+                new Flower("Magnolia", "white", 6, 45, 30),
+                new Flower("Lilac", "purple", 72, 12, 40),
+                new Flower("Peony", "pink", 13, 60, 20),
+                new Flower("Sunflower", "yellow", 54, 200, 90),
+                new Flower("Rose", "red", 45, 75, 60)
+        );
+
+
+        System.out.println("The list of flowers which name starts with 'M' and contains letter 'a' is: "
+                + startsWithMContainsA(flowerList, "M", "a") + "\n");
+
+        System.out.println("The flower with the least living time is: "
+                + minimumLifeTime(flowerList) + "\n");
+
+        System.out.println("The 'Max' String of the 5 generated is: "
+                + maxOfRandomStrings(5) + "\n");
+
+        System.out.println("The map of random integers and their last 2 numbers: "
+                + mapOfRandomIntegersAndTheirSquaredNumber(5) + "\n");
+
+        System.out.println("How many values of the map are greater than 10: "
+                + howManyAreGreaterThanTen() + "\n");
+
+
+        sortRandomStringsSetInReverseOrder(randomString);
+
+        System.out.println(sortFlowersByHeight(flowerList) + "\n");
+
+
+        System.out.println("How many flowers from the list have even number of petals: "
+                + countOfEvenNumberOfPetals(flowerList) + "\n");
+
+        System.out.println("The shortest String from the set is: " + shortestString() + "\n");
+
+        System.out.println("And last, but not least: *THE OPTIONAL*: ");
+        optionalExample();
+    }
+
+    /**
+     * 11)
+     * I needed to create another set because all the random ones are length 10. For some reason it printed me one of
```

Comment:
because it prints the first one when it compares, check the function that compares.